deeppavlov/evolve.py

```diff
@@ -0,0 +1,322 @@
+"""
+Copyright 2017 Neural Networks and Deep Learning lab, MIPT
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import argparse
+from pathlib import Path
+import sys
+import os
+import json
+from copy import deepcopy
+from subprocess import Popen, PIPE
+import pandas as pd
+
+p = (Path(__file__) / ".." / "..").resolve()
+sys.path.append(str(p))
+
+from deeppavlov.core.common.errors import ConfigError
+from deeppavlov.models.evolution.evolution_param_generator import ParamsEvolution
+from deeppavlov.core.common.file import read_json, save_json
+from deeppavlov.core.common.log import get_logger
+from deeppavlov.core.commands.utils import set_deeppavlov_root, expand_path
+
+log = get_logger(__name__)
+
+parser = argparse.ArgumentParser()
+
+parser.add_argument("config_path", help="path to a pipeline json config", type=str)
+parser.add_argument('--key_main_model', help='key inserted in dictionary of main model in pipe', default="main")
+parser.add_argument('--p_cross', help='probability of crossover', type=float, default=0.2)
+parser.add_argument('--pow_cross', help='crossover power', type=float, default=0.1)
+parser.add_argument('--p_mut', help='probability of mutation', type=float, default=1.)
+parser.add_argument('--pow_mut', help='mutation power', type=float, default=0.1)
+
+parser.add_argument('--p_size', help='population size', type=int, default=10)
+parser.add_argument('--gpus', help='visible GPUs divided by comma <<,>>', default="-1")
+parser.add_argument('--train_partition',
+                    help='partition of splitted train file', default=1)
+parser.add_argument('--start_from_population',
+                    help='population number to start from. 0 means from scratch', default=0)
+parser.add_argument('--path_to_population',
+                    help='path to population to start from', default="")
+parser.add_argument('--elitism_with_weights',
+                    help='whether to save elite models with weights or without', default=0)
```

Comment:
maybe use `action='store_true'` for `elitism_with_weights`?

Reply:
I don't think so, because all of the files are actually stored. this parameter is responsible for translating elite models with or without weights from population to popultation

Reply:
ok